server/middlewares/categoryRequest.js

```diff
@@ -0,0 +1,88 @@
+import db from '../models/index';
+
+/**
+ * Middleware to check for valid category request
+ * @param {Object} req
+ * @param {Object} res
+ * @param {Object} next
+ *
+ * @returns {Request|Response|*|void|boolean} res
+ */
+export function categoryRequest(req, res, next) {
+  if (req.body.name === undefined || req.body.name === null) {
+    return res.status(400).send({ error: 'Category name is required' });
+  }
+
+  next();
+}
+
+/**
+ * Find category by id
+ * @param  {Integer} id [description]
+ * @return {Object}    [description]
+ */
+function findCategory(id) {
+  return db.Category.findById(id)
+    .then((category) => {
+      if (!category) {
+        return false;
+      }
+      return category;
+    });
+}
+
+/**
+ * Middleware to check if category is valid
+ * @param {Object} req
+ * @param {Object} res
+ * @param {Object} next
+ * @param {int} id
+ *
+ * @returns {Request|Response|*|void|boolean} res
+ */
+function validateCategory(req, res, next, id) {
+  return db.Category.findById(id)
```

Comment:
I think you can use the findCategory method here instead of making another request to the db.